instrumentation/rocketmq-client-4.8/library/src/test/groovy/io/opentelemetry/instrumentation/rocketmq/RocketMqClientTest.groovy

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.rocketmq
+
+import base.BaseConf
+import io.opentelemetery.instrumentation.rocketmq.AbstractRocketMqClientTest
+import io.opentelemetry.instrumentation.test.LibraryTestTrait
+import org.apache.rocketmq.test.listener.rmq.order.RMQOrderListener
+
+class RocketMqClientTest extends AbstractRocketMqClientTest implements LibraryTestTrait {
+
+  @Override
+  void configureMQProducer() {
+    producer=BaseConf.getProducer(BaseConf.nsAddr)
+    producer.getDefaultMQProducerImpl().registerSendMessageHook(new TracingSendMessageHookImpl())
```

Comment:
We want the pattern to be that as much logic as possible is in the parent class, library instrumentation only registers tracing, while javaagent test usually has an empty body. I think it should be possible if you provide producer and consumer as method arguments to these abstract methods